src/components/NoticeAndDoNotSell.tsx

```diff
@@ -14,24 +15,34 @@ import Title from './Title';
 
 // eslint-disable-next-line jsdoc/require-returns, jsdoc/require-param
 /**
- * Component showing "accept all" interface
+ * Component showing 'okay' button for "do not sell my personal information" interface
  */
 export default function NoticeAndDoNotSell({
   handleSetViewState,
 }: {
   /** Function to change viewState */
   handleSetViewState: HandleSetViewState;
 }): JSX.Element {
+  const { airgap } = useAirgap();
   const { formatMessage } = useIntl();
 
+  const handleConfirm: JSX.MouseEventHandler<HTMLButtonElement> | undefined = (
+    event: JSX.TargetedEvent<HTMLButtonElement, MouseEvent>,
+  ): void => {
+    event.preventDefault();
+    // Confirm current consent
+    airgap.setConsent(event, airgap.getConsent().purposes);
```

Comment:
Does this work? Do you not need to do `event.nativeEvent` ?

Reply:
I tested this and these are the native event objects in this case.